components/src/dynamo/frontend/main.py

```diff
@@ -207,13 +209,32 @@ def parse_args():
         default=False,
         help="Start KServe gRPC server.",
     )
+    parser.add_argument(
+        "--nim-metrics-polling-interval-seconds",
+        type=float,
+        default=float(os.environ.get(NIM_METRICS_POLLING_INTERVAL_ENV_VAR, "0")),
+        help=f"Interval in seconds for polling NIM backend metrics. Set to > 0 to enable polling (default: 0=disabled, suggested: 9.2s which is less than typical Prometheus scrape interval). Cannot be used with --nim-metrics-on-demand. Can be set via {NIM_METRICS_POLLING_INTERVAL_ENV_VAR} env var.",
+    )
+    parser.add_argument(
+        "--nim-metrics-on-demand",
+        action="store_true",
+        default=os.environ.get(NIM_METRICS_ON_DEMAND_ENV_VAR, "false").lower()
+        in ("true", "1", "yes", "y"),
```

Comment:
if we have more than one use for this- would recommend a constant "BOOL_TRUTH_VALUES" or similar